doc/Syscalls.md

```diff
@@ -157,8 +157,9 @@ A callback function is uniquely identified by the pair (`driver`,
 are any pending callbacks for this callback ID, they are removed from the queue
 before the new callback function is bound to the callback ID.
 
-Passing a null pointer callback function disables a previously set callback
-(besides flushing pending callbacks for this callback ID).
+A process can pass a null pointer as the callback argument to request the driver
+disable a previously set callback (besides flushing pending callbacks for this
+callback ID).
```

Comment:
```suggestion
A process can pass a null pointer as the callback argument to request the driver
disable a previously set callback. Note that it is possible for a pending callback
to be issued during the attempt to disable a callback. The callback is not disabled
until the call setting it to null returns.
```

Reply:
I'm not sure that is entirely accurate. Does something prevent the following sequence of events? (I'm not sure, as I haven't touched this code much)

1. A process calls `subscribe` to set a callback.
1. The driver triggers the callback. The callback is enqueued in the process' callback queue.
1. The process passes a null pointer to `subscribe` to disable the callback, and the driver correctly stops queueing callbacks for that subscription ID.
1. The process calls `yield` and the previously-queued callback runs.

Reply:
Mhm, unfortunately I think you are correct. That is currently possible.

Again, I think I projected here what I would _want_ the kernel to do rather than what it does. My concern with the current interface is that it becomes unreasonable to write predictable apps. I think I would like to see the kernel handle this as described, but for now, what you wrote is accurate.

Reply:
+1 on "what I want the kernel to do != what the kernel does". I think that discussion is out of the scope of this PR, however, as changing how subscriptions work is not something that can be done quickly, and we should correct the documentation in the interim.